adventure/adventure.py

```diff
@@ -3068,71 +3087,72 @@ def E(t: str) -> str:
             god = await self.config.guild(self.bot.get_guild(guild_id)).god_name()
         msg = ""
         glyphs = False
-        glyphs_bonus = 0                       
+        glyphs_bonus = 0
+        pray_list_name = []
         for user in pray_list:
+            pray_list_name.append(self.E(user.display_name))
+        total_size = len(fight_list + talk_list + magic_list)
+        for user in magic_list:
             try:
                 c = await Character._from_json(self.config, user)
             except Exception:
                 log.error("Error with the new character sheet", exc_info=True)
                 continue
-            bonus_cleric = 0                   
-            if c.heroclass["name"] == "Cleric":
-                bonus_cleric = int((c.int + c.skill["int"] + c.att + c.skill["att"] + c.cha + c.skill["cha"])/3)
+            if c.heroclass["name"] == "Wizard" and not glyphs and len(pray_list) >= 1:
+                glyphs_power = (c.int + c.skill["int"] + c.cha + c.skill["cha"]) / 2
+                glyphs_chance = min(int(glyphs_power / 2.5 + 1), c.lvl)
+                glyphs_roll = random.randint(1, 100)
+                if glyphs_roll in range(1, glyphs_chance):
+                    glyphs = True
+                    glyphs_bonus = int(glyphs_chance * 0.4) + 3
+                    msg += f"{bold(self.E(user.display_name))}'s magic glyphs start glowing, amplifying all prayers! 🛐+{glyphs_bonus}%.\n"
+        for user in pray_list:
+            try:
+                c = await Character._from_json(self.config, user)
+            except Exception:
+                log.error("Error with the new character sheet", exc_info=True)
+                continue
+            pray_bonus = int((c.int + c.skill["int"] + c.att + c.skill["att"] + c.cha + c.skill["cha"])/3)
             roll = random.randint(1, 20)
-            if len(fight_list + talk_list + magic_list) == 0 or roll == 1: #fumble in case no one is fighting, talking, using magic or if roll 1
-                if len(fight_list + talk_list + magic_list) == 0:
-                    msg += (
-                        f"{bold(self.E(user.display_name))} blessed like a "
-                        "madman but nobody was there to receive it.\n"
-                    )
-                elif c.heroclass["name"] == "Cleric" and c.heroclass["ability"]: #small bonus if the user is cleric with activated ability
+            pray_score = pray_bonus + roll
+            if total_size == 0 or roll == 1: #fumble in case no one is fighting, talking, using magic or if roll 1
+                if total_size == 0:
+                    msg += f"{bold(humanize_list(pray_list_name))} blessed like "
+                    if len(pray_list) == 1:

```

Comment:
could be simplified to something like:
`attrib = "a madman" if len(pray_list) == 1 else "madmen"`
then use attrib inside a single f string

Reply:
Sure why not

Reply:
done